test/FunctionalTests/Framework/TestExecutionLayer/AstoriaDatabase.cs

```diff
@@ -173,7 +173,10 @@ public string DatabaseName
             {
                 if (_databaseName == null)
                 {
-                    _databaseName = String.Format("{0}_{1}_{2}", this.DatabasePrefixName, System.Net.Dns.GetHostName(), Guid.NewGuid().ToString("N"));
+                    _databaseName = String.Format("{0}_{1}_{2}",
+                        this.DatabasePrefixName,
+                        System.Net.Dns.GetHostName().Replace('-', '_'),

```

Comment:
Looks like the only special character allowed in the hostname is the '-', so I'll leave my change as-is.